src/diagnostics.rs

```diff
@@ -671,6 +685,79 @@ impl DiagnosticReporter for CodeSpanDiagnosticReporter {
     }
 }
 
+/// a DiagnosticReporter that reports diagnostics using clang-format
+pub struct ClangFormatDiagnosticReporter {
+    files: SimpleFiles<String, String>,
+}
+
+impl ClangFormatDiagnosticReporter {
+    fn new() -> Self {
+        ClangFormatDiagnosticReporter {
+            files: SimpleFiles::new(),
+        }
+    }
+}
+
+impl Default for ClangFormatDiagnosticReporter {
+    fn default() -> Self {
+        ClangFormatDiagnosticReporter::new()
+    }
+}
+
+impl DiagnosticReporter for ClangFormatDiagnosticReporter {
+    fn report(&self, diagnostics: &[AssessedDiagnostic], file_id: usize) {
+        for ad in diagnostics {
+            let d = &ad.diagnostic;
+            let location = d.get_location();
+            let file_name = match self.files.get(file_id) {
+                Ok(v) => v.name(),
+                Err(err) => {
+                    eprintln!("Unable to find affected file : {}", err);
+                    "unknown"
+                }
+            };
+
+            let start_location = get_location(&self.files, file_id, location.get_start());
+            let end_location = get_location(&self.files, file_id, location.get_end());
+
+            // print diagnostic
+            // file-name:{range}: severity: msg
+            eprintln!(
+                "{}:{{{}:{}-{}:{}}}: {}: {}",
+                file_name,
+                start_location.line_number,
+                start_location.column_number,
+                end_location.line_number,
+                end_location.column_number,
+                ad.severity,
+                d.get_message()
+            );
+        }
+    }
+    fn register(&mut self, path: String, src: String) -> usize {
+        self.files.add(path, src)
+    }
+}
+
+/// returns location at the given byte index in file
+/// on error location with line and colum number 0 will be returned
+fn get_location(
```

Comment:
I think if location is wrong, we should not return 0, we can return either an Err or an Option<Location> here, and we can skip the location part in the error message (just show the file).
Also if the error does not have a range (start = end) we should probably not show a range